File: 4115-minimum-distance-between-three-equal-elements-i/minimum-distance-between-three-equal-elements-i.cpp
class Solution {
public:
    int minimumDistance(vector<int>& nums) {
        int ans = 1e9;
        int n = nums.size();
        for(int i = 0; i<n; i++){
            for(int j = 0; j<n; j++){
                for(int k = 0; k<n; k++){
                    if(i != j && j != k && k != i && nums[i] == nums[j] && nums[j] == nums[k]){
                        ans = min(ans, abs(i-j)+abs(j-k)+abs(k-i));
                    }
                }
            }
        }
       return ans == 1e9 ? -1 : ans;
        
    }
};

File: 4115-minimum-distance-between-three-equal-elements-i/minimum-distance-between-three-equal-elements-i.java
class Solution {
    public int minimumDistance(int[] nums) {
        int n = nums.length;
        int ans = (int)1e9; 
        for(int i = 0; i<n; i++){
            for(int j = 0; j<n; j++){
                for(int k = 0; k<n; k++){
                    if((i != j && j != k && k != i) && (nums[i] == nums[j])  && (nums[j] == nums[k])  && (nums[k] == nums[i])){
                        ans = Math.min(ans, Math.abs(i-j) + Math.abs(j-k) + Math.abs(k-i));
                    }
                }
            }
        }
        return ans == (int) 1e9 ? -1 : ans;
    }
}